website/docs/test_core_lib.md

```diff
@@ -0,0 +1,231 @@
+from core_lib.data_transform.result_to_dict import ResultToDict
+from core_lib.data_layers.service.service import Service
+from user_core_lib.data_layers.data_access.user_data_access import UserDataAccess
+
+
+class UserService(Service):
+    def __init__(self, data_access: UserDataAccess):
+        self.data_access = data_access
+
+    @ResultToDict()
+    def create(self, user_data: dict):
+        return self.data_access.create(user_data)
+
+    @ResultToDict()
+    def get(self, user_id: int):
+        return self.data_access.get(user_id)
+
+    def update(self, user_id: int, update: dict):
+        return self.data_access.update(user_id, update)
+
+    def delete(self, user_id: int):
+        return self.data_access.delete(user_id)
+```
+
+## Main Class
+Here you'll have all the `DataAccess`, `Service`,  `Connection`, `Cache` initialized. Which can be further accessed when we initialize the `Core-Lib`.
+
+### Config
+`user_core_lib.yaml`
+```yaml
+# @package _global_
+core_lib:
+  user_core_lib:
+    data:
+      userdb:
+        _target_: core_lib.connection.sql_alchemy_connection_registry.SqlAlchemyConnectionRegistry
+        config:
+            log_queries: false
+            create_db: true
+            session:
+                pool_recycle: 3200
+                pool_pre_ping: false
+            url:
+                file: ${oc.env:USERDB_DB}
+                protocol: postgresql
+                username: ${oc.env:USERDB_USER}
+                password: ${oc.env:USERDB_PASSWORD}
+                port: ${oc.env:USERDB_PORT}
+                host: ${oc.env:USERDB_HOST}
+    cache:
+        memory_cache:
+            _target_: core_lib.cache.cache_handler_ram.CacheHandlerRam
+    client:
+      user_client:
+        _target_: user_core_lib.UserCoreLib.UserClient
+        base_url: https://example.com/
+```
+
+`user_core_lib.py`
+```python
+from omegaconf import DictConfig
+
+from core_lib.core_lib import CoreLib
+from core_lib.connection.sql_alchemy_connection_registry import SqlAlchemyConnectionRegistry
+from core_lib.helpers.config_instances import instantiate_config
+
+from user_core_lib.data_layers.data_access.customer_data_access import CustomerDataAccess
+from user_core_lib.data_layers.data_access.user_data_access import UserDataAccess
+from user_core_lib.data_layers.service.customer_service import CustomerService
+from user_core_lib.data_layers.service.user_service import UserService
+
+
+class UserCoreLib(CoreLib):
+    def __init__(self, conf: DictConfig):
+        super().__init__()
+        self.config = conf
+        CoreLib.cache_registry.register("memory_cache", instantiate_config(self.config.core_lib.user_core_lib.cache.memory_cache))
+        db_session = instantiate_config(self.config.core_lib.user_core_lib.data.userdb)
+        self.user = UserService(UserDataAccess(db_session))
+        self.user_client = instantiate_config(self.config.core_lib.user_core_lib.client.user_client)
+
+    class UserClient(ClientBase):
```

Comment:
define this class `class UserClient(ClientBase):` above the `UserCoreLib`

we use to define classes like this independently